app-drawer/app-drawer.html

```diff
@@ -513,8 +553,22 @@
         this.translate3d(x + 'px', '0', '0', this.$.contentContainer);
       },
 
-      _updateDocScroll: function() {
-        document.body.style.overflow = (this.persistent || !this.opened) ? '' : 'hidden';
+      _resetDrawerTranslate: function() {
+        this.$.scrim.style.opacity = '';
+        this.transform('', this.$.contentContainer);
+      },
+
+      _resetFocusedContent: function() {
+        if (this._isAboveContent()) {
+          document.body.style.overflow = 'hidden';
+          document.addEventListener('focus', this._boundFocusHandler, true /* useCapture */);
+          this._focusInsideDrawer();
+        } else {
+          document.removeEventListener('focus', this._boundFocusHandler, true /* useCapture */);
+          document.body.style.overflow = '';
+        }
+
+        this.fire('app-drawer-focus-changed');
```

Comment:
Is this needed?  If so we need to add doc for this event.  Also is this also a signal we can use for "end of drawer animation/transition", if so maybe we need a better name for this.